apiserver/facades/agent/storageprovisioner/storageprovisioner.go

```diff
@@ -1398,6 +1404,14 @@ func (s *StorageProvisionerAPIv4) CreateVolumeAttachmentPlans(ctx context.Contex
 		if !canAccess(machineTag, volumeTag) {
 			return apiservererrors.ErrPerm
 		}
+		// Check that the machine is provisioned.
```

Comment:
These seem like new checks, I can't tell what they're replacing? How come they didn't need to be checked in 3.x

Reply:
Yeah, they seem redundant. Or is is the equivalent of not preovisioned? If that's the case, you should have a method IsMachineProvisioned.

Reply:
So, this answers @Aflynn50's concerns in https://github.com/juju/juju/pull/18165#discussion_r1786369674 and https://github.com/juju/juju/pull/18165#discussion_r1786370286. 
What I've basically done is to lift out the checks from the legacy state, one step up on the stack. 
So to answer, no we don't need any jira task because it's done right here in the apiserver and it's instance data's concern.